### DISCUSSION TITLE
Support for merged cells in markdown?

### DISCUSSION THREAD

**Discussion by sodiumchl** [quarto-dev/quarto-cli#2603] [Q&A]
How can I render merged cells correctly? I use ```quarto render test.md --html``` but the merged cells are not merged.

An example below from
https://stackoverflow.com/questions/46621765/can-i-merge-table-rows-in-markdown

```
|             |          Grouping           ||
First Header  | Second Header | Third Header |
 ------------ | :-----------: | -----------: |
Content       |          *Long Cell*        ||
Content       |   **Cell**    |         Cell |

New section   |     More      |         Data |
And more      | With an escaped '\|'         ||  
[Prototype table]
```

**Comment by rgaiacs** [comment]
Hi @sodiumchl and thanks for your interest in Quarto.

Quarto uses Pandoc to convert the `.md` file to `.html`. Pandoc supports many table formats, see https://pandoc.org/MANUAL.html#tables, but

> Pandoc does not support grid tables with row spans or column spans.

**Reply by sodiumchl** [reply]
> Pandoc does not support grid tables with row spans or column spans.

That's an unfortunate limitation. Is there any way to use a different md to html converter instead of pandoc, e.g., multimarkdown?

**Reply by rgaiacs** [reply]
If you use Quarto, you are lock with Pandoc. Pandoc can be extended with filters, see https://pandoc.org/filters.html but I'm not sure if you will be able to get merged cells using filters.

If you are only targeting HTML output, you should have a look at https://renkun-ken.github.io/formattable/.

**Reply by vtraag** [reply]
Perhaps this could use an update. From what I see now, grid tables, such as described in https://pandoc.org/MANUAL.html#tables are fully supported. For example, this example 

```
+---------------+---------------+--------------------+
| Fruit         | Price         | Advantages         |
+===============+===============+====================+
| Bananas       | $1.34         | - built-in wrapper |
|               |               | - bright color     |
+---------------+---------------+--------------------+
| Oranges       | $2.10         | - cures scurvy     |
|               |               | - tasty            |
+---------------+---------------+--------------------+
```

renders perfectly using Quarto to

![image](https://github.com/quarto-dev/quarto-cli/assets/6057804/e85190b5-5e2d-4ec2-85b1-e0c2707caf22)

**Reply by rgaiacs** [reply]
Dear @vtraag, Thanks for your message. Your example has cells with a list. The question is a cell spanning multiple columns or rows (also called merged cells). The correct example from Pandoc documentation is


```
+---------------------+----------+
| Property            | Earth    |
+=============+=======+==========+
|             | min   | -89.2 °C |
| Temperature +-------+----------+
| 1961-1990   | mean  | 14 °C    |
|             +-------+----------+
|             | max   | 56.7 °C  |
+-------------+-------+----------+
```

**Comment by stanstrup** [comment]
the solution by @rgaiacs is broken by RStudio's visual mode.

**Reply by mcanouil** [reply]
Visual Editor uses Pandoc.
So if you use something not supported, it won't work.

This being said, could you elaborate? What is broken exactly? In which context? Which editor?

**Reply by cderv** [reply]
> broken by RStudio's visual mode.

Possibly an issue to report in RStudio IDE on how grid tables are supported in markdown input. 

Edit: or directly at https://github.com/quarto-dev/quarto which is the repo for the visual editor code source

**Reply by stanstrup** [reply]
If you copy in that code in source mode it works. But as soon as you switch to visual mode it "repairs" the table to have no spanned rows or columns.

**Reply by mcanouil** [reply]
> that code

What code?

**Reply by stanstrup** [reply]
the code by @rgaiacs as I wrote:

```
+---------------------+----------+
| Property            | Earth    |
+=============+=======+==========+
|             | min   | -89.2 °C |
| Temperature +-------+----------+
| 1961-1990   | mean  | 14 °C    |
|             +-------+----------+
|             | max   | 56.7 °C  |
+-------------+-------+----------+
```

Put that in the editor in source mode. Switch to visual mode and back to source mode and it as "repaired" the spannning.

**Reply by mcanouil** [reply]
I don't think Visual Editor supports Grid table.
I suggest you look if this feature request was already reported and if not, open a new one with a fully reproducible example, the steps to reproduce the current behaviour and explain what you want/expect on the appropriate repository which is <https://github.com/quarto-dev/quarto>.

**Comment by pmoreno-DTP** [comment]
Any update on this issue? I get the same problem and would love to have a fix.

**Reply by cderv** [reply]
> I get the same problem and would love to have a fix.

What is your problem exactly ? What is the table you are trying to write ? 
Does same problem relates to visual editor support ? or Quarto rendering ? 

Thanks for the additional information

**Comment by pmoreno-DTP** [comment]
Here is a simple table with merged cells that I would like to replicate in Quarto using the visual editor support:

<img width="687" height="652" alt="Screenshot 2026-02-10 111353" src="https://github.com/user-attachments/assets/6333d66d-e5f6-4be9-95cb-2f0e954e77e9" />

I need tables with merged cells, custom borders, background colors, and specific formatting, and they should work well in HTML, PDF, and Word outputs. Would using **flextable** be the best approach for this? Are there simpler or better alternatives I should consider? Any guidance would be appreciated.

**Reply by mcanouil** [reply]
> [!IMPORTANT]
> Accessibility: To improve accessibility, please add alternative text to your screenshots. This helps all users, including those using screen readers, to understand the context of the images. A brief description can make a big difference! See [Good Alt Text, Bad Alt Text — Making Your Content Perceivable](https://www.wcag.com/blog/good-alt-text-bad-alt-text-making-your-content-perceivable/).

---

I'm not sure the Visual Editor is going to be a full editor like Excel or Word. I don't think that's the goal.

Quarto 1.9 brings native support to [`list-table`](https://github.com/pandoc-ext/list-table) which help make markdown table with merged cells.
